medusa/providers/nzb/binsearch.py

```diff
@@ -41,6 +52,230 @@ def __init__(self):
+            'adv_age': '',
+            'xminsize': 20,
+            'max': 250,
+        }
+        groups = [1, 2]
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+                search_params['q'] = search_string
+                for group in groups:
+                    # Try both 'search in the most popular groups' & 'search in the other groups' modes
+                    search_params['server'] = group
+                    if mode != 'RSS':
+                        log.debug('Search string: {search}', {'search': search_string})
+
+                    response = self.get_url(self.urls['search'], params=search_params)
+                    if not response:
+                        log.debug('No data returned from provider')
+                        continue
+
+                    results += self.parse(response.text, mode)
+
+        return results
+
+    def parse(self, data, mode, query=''):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+
+        :return: A list of items found
+        """
+
+        def process_column_header(td):
+            return td.get_text(strip=True).lower()
+
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            table = html.find('table', class_='xMenuT')
+            rows = table('tr') if table else []
+            row_offset = 2
+            if not rows or not len(rows) - row_offset:
+                log.debug('Data returned from provider does not contain any torrents')
+                return items
+
+            headers = rows[0]('th')
+            # 0, 1, subject, poster, group, age
+            labels = [process_column_header(header) or idx
+                      for idx, header in enumerate(headers)]
+
+            # Skip column headers
+            rows = rows[row_offset:]
+
+            for row in rows:
+                try:
+                    col = dict(zip(labels, row('td')))
+                    nzb_id_input = col[1].find('input')
+                    if not nzb_id_input:
+                        continue
+                    nzb_id = nzb_id_input['name']
+                    title_field = col['subject'].find('span')
+                    # Try and get the the article subject from the weird binsearch format
+                    title = BinSearchProvider.title_regex.search(title_field.text).group(1)
+                    for extension in ('.nfo', '.par2', '.rar', '.zip', '.nzb'):
+                        # Strip extensions that aren't part of the file name
+                        title = title.rstrip(extension)
+                except AttributeError as error:
+                    log.debug('Parsing rows, that may not always have usefull info. Skipping to next.')
+                    continue
+                except Exception as error:
```

Comment:
Ooh yeah, need to remove this also. Was to check if any other exceptions pass through.